pkg/generate/app/cmd/newapp.go

```diff
@@ -1109,11 +1110,38 @@ func optionallyValidateExposedPorts(config *AppConfig, repositories app.SourceRe
 	return nil
 }
 
-func exposedPortsAreNumeric(node *dockerfileparser.Node) error {
+func exposedPortsAreValid(node *dockerfileparser.Node) error {
+	allErrs := make([]error, 0)
+
 	for _, port := range dockerfileutil.LastExposedPorts(node) {
-		if _, err := strconv.ParseInt(port, 10, 32); err != nil {
-			return fmt.Errorf("could not parse %q: must be numeric", port)
+		errs := make([]string, 0)
+
+		if strings.HasPrefix(port, "$") {
+			errs = append(errs, "args are not currently supported for port numbers, did you mean to use config.AllowNonNumericExposedPorts?")
```

Comment:
i don't know what config.AllowNonNumericExposedPorts is or how to set it.  How will our users?

Reply:
(I don't think they even can. it's an internal setting we set to true when invoking new-build)

Reply:
updated